### PR TITLE
fix(fromOpenApi): remove extra whitespace from content type

### LINKED ISSUE #61: `fromOpenApi` mishandling `Accept` header
## Description

The accept header can contain whitespace in addition to ',' between content types, if the content schema does not match the first content type then it doesn't match at all.

## Reproduction steps

- Make an openAPI content handler for `*/*`
- Send a request with an accept header with multiple content types: `Accept: text/html, */*`
- Observe that while it responds, the response won't have a body.

## Expected behavior

The content returns as described by the openapi spec.

## Initial assessment

In https://github.com/mswjs/source/blob/main/src/open-api/utils/open-api-utils.ts `toHeaders` and `toBody` split the accept header using `.split(',')`. This means that all content-types besides the first are prefixed by existing whitespace in the accept header.
This then gets included in the regex, which makes it fail to match since the openapi content types usually don't contain whitespace.

Currently I'm patching this by adding `.trim()` to the `contentTypeToRegExp` method.

### PR DESCRIPTION
**Pull request by Kiskae**
Fixes #61

### CODE REVIEW

File: src/open-api/utils/open-api-utils.ts [hunk 1]
```diff
@@ -263,6 +257,14 @@ export function toBody(
   return null
 }
 
+function getAcceptedContentTypes(request: Request): string[] {
```
Comment (kettanaito) [inline src/open-api/utils/open-api-utils.ts:260]: Love this!

**Review by kettanaito — APPROVED** [review]
Thank you for jumping on this so fast, @Kiskae! The changes look great. The only thing that's missing is an automated test for this. 

Feel free to explore and add one, otherwise I will try to add it later next week. Thank you!

**Comment by weyert** [comment]
@Kiskae Thanks for taking care of this. Never thought of the whitespaces 🤗 
Personally, barely ever pass multiple content-types to `Accept`

**Review by weyert — APPROVED** [review]
LGTM

**Comment by Kiskae** [comment]
> Thank you for jumping on this so fast, @Kiskae! The changes look great. The only thing that's missing is an automated test for this.
> 
> Feel free to explore and add one, otherwise I will try to add it later next week. Thank you!

Added some tests. After looking at the RFC I realized that there are more complex accept headers that aren't being handled, but solving that probably involves adding dependencies. So I've just added some skipped tests as a reminder that they exist.

Also made it so the automated tests in src actually get executed :P

**Comment by kettanaito** [comment]
This looks fantastic! Thank you for adding the tests, @Kiskae!

I've polished the things a bit, mostly around test names and changing the `getAcceptedContentTypes` to accept just the `Headers` instance (it never needed anything else from the request). Made it easier to test too.

**Comment by kettanaito** [comment]
## Released: v0.3.1 🎉

This has been released in v0.3.1!

- 📄 [**Release notes**](https://github.com/mswjs/source/releases/tag/v0.3.1)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/source/v/0.3.1)

Make sure to always update to the latest version (`npm i @mswjs/source@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**